src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -473,6 +473,17 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 			}
 			return findWidget;
 		});
+		const viewKey = new Set<string>();
+		viewKey.add('focusedView');
+		this._register(this.onDidFocus(() => TabFocus.setTabFocusMode(this._configurationService.getValue(TerminalSettingId.TabFocusMode))));
+		this._register(contextKeyService.onDidChangeContext((c) => {
+			if (c.affectsSome(viewKey)) {
+				const view = contextKeyService.getContextKeyValue('focusedView');
```

Comment:
record what the value was before setting it and any time the setting changes, just clear the value